### PR TITLE
Update quote & syn dependencies to 1.x

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Updates the quote & syn dependencies to 1.x

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch/src/http/transport.rs [hunk 1]
```diff
@@ -49,6 +49,7 @@ use crate::{
 };
 use base64::write::EncoderWriter as Base64Encoder;
 use bytes::BytesMut;
+use dyn_clone::clone_trait_object;
```
Comment (VachaShah) [inline opensearch/src/http/transport.rs:52]: Is this used anywhere?

Comment (Xtansia) [reply opensearch/src/http/transport.rs:52]: Yeah, it's used at line 493: https://github.com/opensearch-project/opensearch-rs/blob/35da3730f719b9b4b8ccdf5336c8943561725cba/opensearch/src/http/transport.rs#L488-L493

**Review by VachaShah — APPROVED** [review]
LGTM!

**Comment by VachaShah** [comment]
@dblock Review please?

**Review by dblock — APPROVED** [review]

**Pull request merged**